### ISSUE TITLE
[Feature] Multi language config

### ISSUE THREAD

**Issue by frank-zsy** [hypertrons/hypertrons-crx#143]
As far as I know, right now the project support multi language but no config entry found in the config page.

We may need to expose the function to change language of the extension.

**Comment by heming6666** [comment]
As far as I know, the displayed language is determined by `Chrome` and we can't change it in code.

@LiuChangFreeman Is that right?

**Comment by LiuChangFreeman** [comment]
> As far as I know, the displayed language is determined by `Chrome` and we can't change it in code.
> 
> @LiuChangFreeman Is that right?

Since the display language is determined by `getMessageI18n`, we can manually change it in the code in some way.   
There should be some survey into it.

**Comment by LiuChangFreeman** [comment]
This may be a suitable method to force display language in Chinese.   
All other languages can be config in this way.   
```js
import message_en from '../_locales/en/messages.json'
import message_cn from '../_locales/zh_CN/messages.json'

export function getMessageI18n(key: string) {
  // @ts-ignore
  return message_cn[key]["message"]
}
```

**Comment by heming6666** [comment]
> This may be a suitable method to force display language in Chinese.
> All other languages can be config in this way.

I think it's a good way. We can get the default language used by chrome and alse add config entry in the config page.

**Comment by LiuChangFreeman** [comment]
Here is a problem:  
The only  `sync` storage api is `localStorage` but it can't be used across different pages.  
If we want to config display language, the `getMessageI18n` must be rewrite into this version:  
```ts
import messages_en from '../_locales/en/messages.json';
import messages_zh_CN from '../_locales/zh_CN/messages.json';

const messages_locale={
  "en":messages_en,
  "zh_CN":messages_zh_CN
}

export function getMessageI18n(key: string,locale: string) {
  // @ts-ignore
  return messages_locale[locale][key]["message"];
}
``` 
We must add another parameter `locale` in each page where we use `getMessageI18n`.

**Comment by LiuChangFreeman** [comment]
> Here is a problem:
> The only `sync` storage api is `localStorage` but it can't be used across different pages.
> If we want to config display language, the `getMessageI18n` must be rewrite into this version:
> 
> ```ts
> import messages_en from '../_locales/en/messages.json';
> import messages_zh_CN from '../_locales/zh_CN/messages.json';
> 
> const messages_locale={
>   "en":messages_en,
>   "zh_CN":messages_zh_CN
> }
> 
> export function getMessageI18n(key: string,locale: string) {
>   // @ts-ignore
>   return messages_locale[locale][key]["message"];
> }
> ```
> 
> We must add another parameter `locale` in each page where we use `getMessageI18n`.

Do you think there is another better way? @heming6666

**Comment by heming6666** [comment]
> Here is a problem:
> The only `sync` storage api is `localStorage` but it can't be used across different pages.
> If we want to config display language, the `getMessageI18n` must be rewrite into this version:
> 
> ```ts
> import messages_en from '../_locales/en/messages.json';
> import messages_zh_CN from '../_locales/zh_CN/messages.json';
> 
> const messages_locale={
>   "en":messages_en,
>   "zh_CN":messages_zh_CN
> }
> 
> export function getMessageI18n(key: string,locale: string) {
>   // @ts-ignore
>   return messages_locale[locale][key]["message"];
> }
> ```
> 
> We must add another parameter `locale` in each page where we use `getMessageI18n`.

I think this is a quite good way.

For `options` and `popup` pages, it's easy to get `locale` parameter from `storage` and then replace the old `getMessageI18n` with the new one.

For `content` pages, we can get `locale` parameter from `storage` in `Perceptor.tsx` and then pass it to every component when loading these components.

I will submit a PR to implement this.

**Comment by LiuChangFreeman** [comment]
> > Here is a problem:
> > The only `sync` storage api is `localStorage` but it can't be used across different pages.
> > If we want to config display language, the `getMessageI18n` must be rewrite into this version:
> > ```ts
> > import messages_en from '../_locales/en/messages.json';
> > import messages_zh_CN from '../_locales/zh_CN/messages.json';
> > 
> > const messages_locale={
> >   "en":messages_en,
> >   "zh_CN":messages_zh_CN
> > }
> > 
> > export function getMessageI18n(key: string,locale: string) {
> >   // @ts-ignore
> >   return messages_locale[locale][key]["message"];
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We must add another parameter `locale` in each page where we use `getMessageI18n`.
> 
> Thanks @LiuChangFreeman . I think this is a quite good way.
> 
> For `options` and `popup` pages, it's easy to get `locale` parameter from `storage` and then replace the old `getMessageI18n` with the new one.
> 
> For `content` pages, we can get `locale` parameter from `storage` in `Perceptor.tsx` and then pass it to every component when loading these components.
> 
> I will submit a PR to implement this.

I can firstly implement the `locale` field in `settings.ts` before anything else.

**Comment by LiuChangFreeman** [comment]
Full support for this feature has been done.

**Comment by heming6666** [comment]
Fixed by #149
